### ISSUE TITLE
Missing 'state' check of existing firewall rule.

### ISSUE THREAD

**Issue by skasai** [puppet-community/puppet-windows_firewall#23]
In looking at the code, I noticed there is also an addition issue that may have not been considered in the windows_firewall that may need to be noted.

There is checks for the rule's existence, but not check if rule not only exists, but if parameters are the same or updated.

This means that the module will not update the rule if there is a change in the parameters.

IE: Port change or IP change.

This may have to be noted that a module will be needed created to remove old settings and applied before pushing new changes.

**Comment by rychannel** [comment]
Yeah, this one bit me today.  Accidentally had a rule set to the wrong port and couldn't update it without manually removing the rule.
